webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/mail/receive/MailCommandAction.java

```diff
@@ -91,22 +134,56 @@ private void processReferenceMailCommands(MimeMessage message, String referenceI
                     bus.publish(new OpenConnectionEvent(wonUri.getUri()));
                     break;
                 case IMPLICIT_OPEN_CONNECTION:
-                    bus.publish(new OpenConnectionEvent(
-                      wonUri.getUri(),  mailContentExtractor.getTextMessage(message)));
+                    bus.publish(new OpenConnectionEvent(wonUri.getUri(), mailContentExtractor.getTextMessage(message)));
                     break;
                 case SENDMESSAGE:
-                    bus.publish(new SendTextMessageOnConnectionEvent(
-                      mailContentExtractor.getTextMessage(message), wonUri.getUri()));
+                    bus.publish(new SendTextMessageOnConnectionEvent(mailContentExtractor.getTextMessage(message), wonUri.getUri()));
+                    break;
+                case CLOSE_NEED:
+                    bus.publish(new NeedDeactivatedEvent(needUri));
                     break;
-
                 case NO_ACTION:
                 default:
                     //INVALID COMMAND
                     logger.error("No command was given or assumed");
                     break;
             }
-        }catch(Exception e){
-            logger.error("no reply mail was set or found");
+        }catch (AccessControlException ace){
+            logger.error("ACCESS RESTRICTION: sender of original and command mail are not equal, command will be blocked");
+        } catch(Exception e){
+            logger.error("no reply mail was set or found: "+e.getMessage());
+        }
+    }
+
+    /**
+     * This Method tries to find a corresponding open need uri from a user(given by the from adress) and returns the
+     * corresponding need uri if there was an open need with the same title
+     * @param message used to extract sender adress and subject(title)
+     * @return
+     */
+    private URI retrieveCorrespondingNeedUriFromMailByTitle(MimeMessage message){
```

Comment:
... and if the subject lines aren't exactly equal, it may be ok to send an email to the user explaining that we wanted to deactivate a need but did not know which one, and she may want to review her needs/mails (but for now just not deactivating anything is ok too)